mlflow/store/artifact/runs_artifact_repo.py

```diff
@@ -128,7 +132,40 @@ def download_artifacts(self, artifact_path, dst_path=None):
         Returns:
             Absolute path of the local filesystem location containing the desired artifacts.
         """
-        return self.repo.download_artifacts(artifact_path, dst_path)
+        try:
+            return self.repo.download_artifacts(artifact_path, dst_path)
+        except Exception:
+            from mlflow.store.artifact.artifact_repository_registry import get_artifact_repository
+
+            _logger.debug(
+                f"Failed to download artifacts from {self.artifact_uri}/{artifact_path}. "
+                "Searching for logged models associated with the run instead."
+            )
+            run_id = RunsArtifactRepository.parse_runs_uri(self.artifact_uri)[0]
+            client = mlflow.tracking.MlflowClient()
+            run = client.get_run(run_id)
+            page_token = None
+            while True:
+                page = client.search_logged_models(
+                    experiment_ids=[run.info.experiment_id],
+                    # TODO: Use filter_string once the backend supports it
+                    # filter_string="...",
+                    page_token=page_token,
+                )
+                for model in page:
+                    # Return the first model that matches the run_id and artifact_path
+                    if model.source_run_id == run_id and model.name == artifact_path:
+                        repo = get_artifact_repository(model.artifact_location)
+                        return repo.download_artifacts(
+                            artifact_path=".",  # root directory
+                            dst_path=dst_path,
+                        )
```

Comment:
The current implementation is:

- Just find the first match
- not sure if there are multiple matching models